solution/infra/lambda.tf

```diff
@@ -0,0 +1,88 @@
+locals {
+  file_name = "../artifacts/cloud-backend-test-lambda.zip"
+  mapbox_api_key = "INSERT_YOUR_KEY_HERE"
+}
+
+# Creates the lambda function using the file specified above
+resource "aws_lambda_function" "gql_lambda_function" {
+  depends_on = [
+    aws_dynamodb_table.users_dynamodb_table
+  ]
+  function_name = var.function_name
+  handler = "lambda-server.graphqlHandler"
+  role = aws_iam_role.gql_lambda_role.arn
+  runtime = "nodejs14.x"
+
+  filename = local.file_name
+  source_code_hash = filebase64sha256(local.file_name)
+
+  environment {
+    variables = {
+      USERS_TABLE_NAME = "${aws_dynamodb_table.users_dynamodb_table.name}"
+      MAPBOX_API_KEY = local.mapbox_api_key
+      DEPLOYMENT_STAGE_NAME = var.deployment_stage
+    }
+  }
+}
+
+# Add IAM role to the created Lambda
+
+resource "aws_iam_role" "gql_lambda_role" {
+  name = "cloud_backend_test_gql_lambda_role"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "lambda.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+}
```

Comment:
Is there a way to avoid raw JSON on Terraform and what you think of keeping JSON in TF?

Reply:
To load the JSON from another file I would use the "file" function, it would avoid raw JSON on Terraform. I really don't like keeping JSON on .tf files, but all Terraform and AWS documentation I saw raw JSON of TF files, just like [here](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lambda_function) and [here](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/iam_role), so I thought it was standard practice. If I had to choose, I would definitely not put raw JSON in TF files.